### PR TITLE
feat: get /me endpoint

### PR DESCRIPTION
**Pull request by johnytiago**
Depends on #84

### CODE REVIEW

File: apps/users-service/src/handlers/fetch-me.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  };
+
+  if (!headers.authorization) {
+    throw Boom.unauthorized();
+  }
+
+  const [type, token] = headers.authorization.split(' ');
+
+  if (type.toLowerCase() !== 'bearer') {
+    throw Boom.forbidden();
```
Comment (jeysal) [inline apps/users-service/src/handlers/fetch-me.ts:25]: Afaik usually Unauthorized is for when you haven't provided a valid token, and Forbidden for when you have provided a valid token, but do not have permissions for the operation. So I think this should be unauthorized instead?

File: apps/users-service/src/handlers/fetch-me.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    got(`https://${authConfig.domain}/userinfo`, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+        'Content-Type': 'application/json',
+      },
+    }).json<auth0.UserInfo>(),
+  );
+
+  if (err) {
+    throw Boom.forbidden('Forbidden', {
```
Comment (jeysal) [inline apps/users-service/src/handlers/fetch-me.ts:38]: Can (and do we want to) forward the status code from auth0 userinfo? I think probably yes - if they give a 4xx it's probably something about the token the client provided, we want to pass that through, if it's 5xx it's probably some Auth0 outage or whatever and we don't want the client to think their authentication is just invalid.

File: apps/users-service/src/handlers/fetch-me.ts (hunk 1 above)
Comment (johnytiago) [reply apps/users-service/src/handlers/fetch-me.ts:25]: Just revisited the HTTP codes spec. You're totally right! 👍

File: apps/users-service/src/handlers/fetch-me.ts (hunk 2 above)
Comment (johnytiago) [reply apps/users-service/src/handlers/fetch-me.ts:38]: That's right, let me add a test to cover that scenario.

File: apps/users-service/test/handlers/users.fetch-me.test.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import { apiGatewayEvent } from '../helpers/events';
+import { APIGatewayProxyResult } from 'aws-lambda';
+import { createRandomUser } from '../helpers/create-user';
+import { config as authConfig } from '@asap-hub/auth';
+
+jest.mock('@asap-hub/auth');
+
+const chance = new Chance();
+
+describe('GET /users/invites/{code}', () => {
```
Comment (fampinheiro) [inline apps/users-service/test/handlers/users.fetch-me.test.ts:13]:
```suggestion
describe('GET /me', () => {
```

File: apps/users-service/test/handlers/users.fetch-me.test.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  test('return 401 when Authentication is not set', async () => {
```
Comment (fampinheiro) [inline apps/users-service/test/handlers/users.fetch-me.test.ts:14]:
```suggestion
  test('return 401 when authorization header is not set', async () => {
```

File: apps/users-service/src/handlers/fetch-me.ts (hunk 2 above)
Comment (johnytiago) [reply apps/users-service/src/handlers/fetch-me.ts]: The fix is done. But I'll need someones help on fixing the typescript error.
image

Comment (jeysal) [reply apps/users-service/src/handlers/fetch-me.ts]: I can TAL (not sure when exactly because of upcoming flight)

Comment (johnytiago) [reply apps/users-service/src/handlers/fetch-me.ts]: I think I managed to work around it 😉 For now it's ok!

**Pull request merged**